### ISSUE TITLE
Crashes when a file is added to shared folder

### ISSUE THREAD

**Issue by kolja** [kolyvan/kybook#345]
TL;DR
when a new file has been added to an SMB Share, and I try to access the device, the app instantly terminates.
--
I keep a some epub and pdf files on a Synology Diskstation in my local network. I keep a reference to the Diskstation on the KyBook main Screen (the one with the round icons). Each time I add a new file to the Diskstation -- when I try to access that file by clicking the diskstation reference, the app crashes.
Once I reopen the app and click on the diskstation reference again, everything is fine and I don't get another crash.
Please fix this. Of course it would be best if the app would just work and not crash. But it would helpful already if there was an error message informing the user what went wrong (e.g "the external SMB Device's directory list has come out of sync, please restart the app"). Also there could be a "refresh" button to avoid having to restart the app altogether (just my 2 cents).

Anyway, thanks for this cool eBook reader. I am using it every day. Keep up the good work!
